aws/ecr/ecr.tf

```diff
@@ -34,7 +32,382 @@ resource "aws_ecr_lifecycle_policy" "form_viewer_policy" {
 }
 
 #
-# Elastic Container Registry
+# Holds the containerized Lambda functions
+#
+
+resource "aws_ecr_repository" "audit_logs_lambda" {
+  name                 = "audit-logs-lambda"
+  image_tag_mutability = "IMMUTABLE"
+
+  image_scanning_configuration {
+    scan_on_push = true
+  }
+}
+
+resource "aws_ecr_lifecycle_policy" "audit_logs_lambda" {
+  repository = aws_ecr_repository.audit_logs_lambda.name
+
+  policy = jsonencode({
+    rules = [{
+      rulePriority = 1
+      description  = "Keep last 30 images"
+
+      selection = {
+        tagStatus     = "tagged"
+        tagPrefixList = ["v"]
+        countType     = "imageCountMoreThan"
+        countNumber   = 30
+      }
+
+      action = {
+        type = "expire"
+      }
+    }]
+  })
+}
+
+resource "aws_ecr_repository" "audit_logs_archiver_lambda" {
+  name                 = "audit-logs-archiver-lambda"
+  image_tag_mutability = "IMMUTABLE"
+
+  image_scanning_configuration {
+    scan_on_push = true
+  }
+}
+
+resource "aws_ecr_lifecycle_policy" "audit_logs_archiver_lambda" {
+  repository = aws_ecr_repository.audit_logs_archiver_lambda.name
+
+  policy = jsonencode({
+    rules = [{
+      rulePriority = 1
+      description  = "Keep last 30 images"
+
+      selection = {
+        tagStatus     = "tagged"
+        tagPrefixList = ["v"]
+        countType     = "imageCountMoreThan"
+        countNumber   = 30
+      }
+
+      action = {
+        type = "expire"
+      }
+    }]
+  })
+}
```

Comment:
Another code style suggestion if you want to simplify the Terraform a bit.  You could use an [array and `for_each`](https://developer.hashicorp.com/terraform/language/meta-arguments/for_each) in this file to generate all the ECRs and their policies:
```hcl
locals {
  ecr_names = toset([
    "audit-logs-archiver-lambda", 
    "nagware-lambda",
    # etc.
  ])
}

resource "aws_ecr_repository" "lambda" {
  for_each = ecr_names

  name                 = each.key
  image_tag_mutability = "IMMUTABLE"

  image_scanning_configuration {
    scan_on_push = true
  }
}

resource "aws_ecr_lifecycle_policy" "lambda_policy" {
  for_each = ecr_names

  repository = aws_ecr_repository.lambda[each.key].name

  policy = jsonencode({
    rules = [{
      rulePriority = 1
      description  = "Keep last 30 images"

      selection = {
        tagStatus     = "tagged"
        tagPrefixList = ["v"]
        countType     = "imageCountMoreThan"
        countNumber   = 30
      }

      action = {
        type = "expire"
      }
    }]
  })
}
```

Reply:
Nice!